app/search/gleaner.py

```diff
@@ -254,6 +254,11 @@ def execute_query(self, page_number):
 
     def text_search(self, **kwargs):
         text = kwargs.pop('text', None)
+        if text!=None:
```

Comment:
Do you need the if / elif here to check whether the text starts or ends with quotes? It seems like you could just call `escape_char` if text isn't None, because `escape_char` is also doing its own checks.

Reply:
That is true. i just tweaked it in the new commit

Reply:
I don't think hat's the greatest idea because what happens if you search for something with an apostrophe in it? Does having an odd number single quotes cause the same problem as an odd number of double quotes?

Reply:
I think you could just escape the single quotes and otherwise leave them alone, but your process for dealing with double quotes looks good.

Reply:
Thats true!